ftp-ballerina/Module.md

```diff
@@ -0,0 +1,214 @@
+## Overview
+
+This module provides an FTP client and an FTP server listener implementation to facilitate an FTP connection to a remote location.
+
+### FTP Client
+
+The `ftp:Client` connects to an FTP server and performs various operations on the files. Currently, it supports the
+generic FTP operations; `get`, `delete`, `put`, `append`, `mkdir`, `rmdir`, `isDirectory`, `rename`, `size`, and
+ `list`.
+
+An FTP client endpoint is defined using the parameters `protocol` and `host`, and optionally the `port` and
+`secureSocket`. Authentication configuration can be configured using the `secureSocket` parameter for basicAuth,
+private key, or TrustStore/Keystore.
+
+##### Creating a client
+
+The following code creates an FTP client and perform I/O operations, which connects to the FTP server with Basic Auth.
+```ballerina
+import ballerina/ftp;
+
+// Define FTP client configuration
+ftp:ClientEndpointConfig ftpConfig = {
+    protocol: ftp:FTP,
+    host: "<The FTP host>",
+    port: <The FTP port>,
+    secureSocket: {
+        basicAuth: {
+            username: "<The FTP username>",
+            password: "<The FTP passowrd>"
+        }
+    }
+};
+
+// Create FTP client
+ftp:Client ftpClient = new(ftpConfig);
+```
+
+##### Creating a directory
+
+The following code creates a directory in the remote FTP server.
+
+```ballerina
+ftp:Error? mkdirResponse = ftpClient->mkdir("<The directory path>");
+if (mkdirResponse is ftp:Error) {
+    log:printError("Error occured in creating directory", err = mkdirResponse);
+    return;
+}
+```
+
+##### Uploading file to a remote server
+
+The following code uploads a file to a remote FTP server.
+
+```ballerina
+io:ReadableByteChannel|error summaryChannel
```

Comment:
This section needs to be updated.

Reply:
Will update with new `stream` based APIs once the code is changed.